core/src/test/java/org/infinispan/health/ClusterHealthImplTest.java

```diff
@@ -1,107 +1,181 @@
+      cacheManager.getCache(CACHE_NAME, true);
+
+      HealthStatus healthStatus = clusterHealth.getHealthStatus();
+
+      assertEquals(HealthStatus.HEALTHY, healthStatus);
+   }
+
+   public void testUnhealthyStatusWhenUserCacheIsStopped() throws Exception {
+      Cache testCache = cacheManager.getCache(CACHE_NAME, true);
+      testCache.stop();
+
+      HealthStatus healthStatus = clusterHealth.getHealthStatus();
+
+      assertEquals(HealthStatus.UNHEALTHY, healthStatus);
+   }
+
+   public void testRebalancingStatusWhenUserCacheIsRebalancing() throws Exception {
+      Cache mockedCache = mock(Cache.class);
+      AdvancedCache mockedAdvancedCache = mock(AdvancedCache.class);
+      DistributionManager mockedDistributionManager = mock(DistributionManager.class);
+      when(mockedCacheManager.getCacheNames()).thenReturn(Collections.singleton(CACHE_NAME));
+
+      mockRehashInProgress(CACHE_NAME, mockedCache, mockedAdvancedCache, mockedDistributionManager);
+
+      ClusterHealth clusterHealth = new ClusterHealthImpl(mockedCacheManager);
+
+      assertEquals(HealthStatus.REBALANCING, clusterHealth.getHealthStatus());
+   }
+
+   public void testHealthyStatusForInternalCaches() throws Exception {
+      cacheManager.getCache(INTERNAL_CACHE_NAME, true);
+
+      assertEquals(HealthStatus.HEALTHY, clusterHealth.getHealthStatus());
+   }
+
+   public void testUnhealthyStatusWhenInternalCacheIsStopped() throws Exception {
+      Cache internalCache = cacheManager.getCache(INTERNAL_CACHE_NAME, true);
+      internalCache.stop();
+
+      assertEquals(HealthStatus.UNHEALTHY, clusterHealth.getHealthStatus());
+   }
+
+   public void testRebalancingStatusWhenInternalCacheIsRebalancing() throws Exception {
+      Cache mockedCache = mock(Cache.class);
+      AdvancedCache mockedAdvancedCache = mock(AdvancedCache.class);
+      DistributionManager mockedDistributionManager = mock(DistributionManager.class);
+
+      when(mockedCacheManager.getCacheNames()).thenReturn(Collections.emptySet());
+      mockRehashInProgress(INTERNAL_CACHE_NAME, mockedCache, mockedAdvancedCache, mockedDistributionManager);
+
+      ClusterHealth clusterHealth = new ClusterHealthImpl(mockedCacheManager);
+
+      assertEquals(HealthStatus.REBALANCING, clusterHealth.getHealthStatus());
+   }
+
+   public void testGetNodeNames() throws Exception {
+      assertTrue(clusterHealth.getNodeNames().get(0).contains(NODE_NAME));
+   }
+
+   public void testGetNumberOfNodes() throws Exception {
+      assertEquals(1, clusterHealth.getNumberOfNodes());
+   }
+
+   public void testGetNumberOfNodesWithNullTransport() throws Exception {
+      ClusterHealth clusterHealth = new ClusterHealthImpl(mockedCacheManager);
+
+      assertEquals(1, clusterHealth.getNumberOfNodes());
+   }
+
+   public void testGetNodeNamesWithNullTransport() throws Exception {
+      ClusterHealth clusterHealth = new ClusterHealthImpl(mockedCacheManager);
```

Comment:
@slaskawi I rewrote the testReturningNodeNamesWithNullTransport

Reply:
Ok, got it. Thanks!